opendevin/runtime/docker/ssh_box.py

```diff
@@ -280,6 +301,37 @@ def __init__(
         atexit.register(self.close)
         super().__init__()
 
+    def _build_sandbox_image(self, base_image):
+        try:
+            if not os.path.exists(AGNOSTIC_SANDBOX_BUILD_DIR):
+                os.makedirs(AGNOSTIC_SANDBOX_BUILD_DIR)
+            dockerfile_content = f"""
+FROM {base_image}
+
+RUN apt update && apt install -y openssh-server wget
+RUN mkdir -p -m0755 /var/run/sshd
+RUN mkdir -p /opendevin && mkdir -p /opendevin/logs && chmod 777 /opendevin/logs
+
+RUN wget "https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-$(uname)-$(uname -m).sh"
+RUN bash Miniforge3-$(uname)-$(uname -m).sh -b -p /opendevin/miniforge3
+RUN bash -c ". /opendevin/miniforge3/etc/profile.d/conda.sh && conda config --set changeps1 False && conda config --append channels conda-forge"
+RUN echo "export PATH=/opendevin/miniforge3/bin:$PATH" >> ~/.bashrc
+            """.strip()
+            with open(f"{AGNOSTIC_SANDBOX_BUILD_DIR}/Dockerfile", "w") as file:
+                file.write(dockerfile_content)
+
+            image, logs = self.docker_client.images.build(path=AGNOSTIC_SANDBOX_BUILD_DIR,
+                                                          tag=AGNOSTIC_SANDBOX_IMAGE_NAME)
```

Comment:
I am concerned that this will cause Docker to launch slowly, as it requires building an image during startup.

Reply:
I think this is ONLY run when user brings their own container -- 
see L235:
`self.use_agnostic_sandbox = "ghcr.io/opendevin/sandbox" not in self.container_image`

Reply:
Get it. Sounds good to me.